app/controllers/pages_controller.rb

```diff
@@ -55,11 +55,11 @@ def destroy
   private
 
   def set_page
-    @page = Page.find(params[:id])
+    @page = Page.find_by(slug: params[:slug_or_id]) || Page.find(params[:slug_or_id])
```

Comment:
find_byはfindも兼ねるので、`@page = Page.find_by(slug: params[:slug_or_id])`だけでいいような気もしたのですが、いかがでしょうか〜(自信はないです)

https://qiita.com/tsuchinoko_run/items/f3926caaec461cfa1ca3

Reply:
`@page = Page.find_by(slug: params[:slug_or_id])`では`slug`をキーに検索している（`id`では検索できていない）ので、`slug`が登録されていないときに`Page.find(params[:slug_or_id])`で`id`からマッチするものを探しています！
リンクの記事では`id`で検索するときは`find_by`でもできるよ〜ってことかなと自分は思いました（`find_by`だとマッチしなかったときに例外が発生しないので`id`の方では`find`を使っています）！

Reply:
すみません、find_byでもIDで検索できると勘違いしていました！

記事を読み直したら書いてありました😅　頓珍漢なことを言ってしまってすみません🙇🏻
勉強になりました。ありがとうございます！！

Reply:
slugはアルファベットから始まるので、以下のように書くと `find_by` でまとめられそうかなと思いました。

```suggestion
    attr_name = params[:slug_or_id]).start_with?(/[a-z]/) ? :slug : :id
    @page = Page.find_by(attr_name => params[:slug_or_id])
```

`Page.find_by(...) || Page.find(...)` だとSQLが2回発行される事があるため、それよりも常に1回のSQLが発行される書き方の方がパフォーマンスが良いです。

Reply:
他でも使いそうなのと、どういうことをやってるのかの説明も兼ねてPageクラスのメソッドにするとスマートかもです〜

Reply:
> slugはアルファベットから始まるので、以下のように書くと `find_by` でまとめられそうかなと思いました。
> 
> `Page.find_by(...) || Page.find(...)` だとSQLが2回発行される事があるため、それよりも常に1回のSQLが発行される書き方の方がパフォーマンスが良いです。

@sinsoku わー！見逃してました💧参考になります！ありがとうございます！